src/schema_compile_node.c

```diff
@@ -1185,10 +1185,8 @@ lys_compile_type_enums(struct lysc_ctx *ctx, struct lysp_type_enum *enums_p, LY_
 
         if (basetype == LY_TYPE_ENUM) {
             if (enums_p[u].flags & LYS_SET_VALUE) {
+                /* value assigned by model */
                 cur_val = (int32_t)enums_p[u].value;
```

Comment:
I'm afraid that here we need to check for collision with the previously (auto-)assigned values.

Reply:
The wrong sequence can be e.g.: 1, 2, - (auto-assigned 3), 3

Reply:
Hi. The check for collision is on line 1191.

Reply:
ok, sorry, I didn't notice